buildconfig/Jenkins/Conda/build

```diff
@@ -65,15 +71,15 @@ if [[ $ENABLE_BUILD_CODE == true ]]; then
 fi
 
 if [[ $ENABLE_UNIT_TESTS == true ]]; then
-    cmake --build . --target AllTests $BUILD_OPTIONS
-fi
-
-if [[ $ENABLE_UNIT_TESTS == true || $ENABLE_SYSTEM_TESTS == true || $ENABLE_DOC_TESTS == true ]]; then
+    # this will get the test data
```

Comment:
If a job has all three of `ENABLE_{UNIT,SYSTEM,DOC}_TESTS` set to true, isn't this adding duplication?

Reply:
Rerunning a target that was already "done" will be a no-op, but it does take time. In principle, `AllTests` depends on `UnitTestData` and `docs-test` depends on `DocTestData`, though there are a fair number of unit tests that use doc test data and vice versa. The idea is to mix in downloading/linking data with the other bits of the build so ninja can download data while other targets are stuck on linking.

Let me refactor this to reduce that more.